Reset the transform of `Box` and `Pallet` instead of just position.

Mareh07 opened the pull request:
Remove the redundant line **Rotation = Vector3.Zero;** in **Box.cs**. 

The rotation is already reset using **rigidBody.Rotation = Vector3.Zero;** earlier in the method. This simplifies the code and avoids confusion.

![image](https://github.com/user-attachments/assets/c3390c09-c278-44a9-a292-b4df52ba9058)

ryevdokimov commented:
It's a little confusing and may have to be reworked completely in the future since it's kind of hack, but the rigid body transform, and its parent node transform are not the same. Where they are depends on whether the simulation is running or not, and if the box is selected or not. When the simulation is not running, the rigid body follows its parent exactly, but when it's running and not selected its free to move independently. This prevents having to recalculate the parent node transforms for no reason.  

With that said. Your modification will prevent the box from returning back to its original rotation before the simulation started.

The code in this file can be cleaned up though. 

```
rigidBody.Position = Vector3.Zero;
rigidBody.Rotation = Vector3.Zero;
rigidBody.Scale = Vector3.One;
````
Can simply become:

`rigidBody.Transform = Transform3D.Identity;`

As well, initialPos can probably just become initalTransform as a Transform3D, and do a similar thing.

`Transform = initalTransform`

cooperra commented on src/Box/Box.cs:
```diff
@@ -113,7 +113,6 @@ void OnSimulationEnded()
 				GlobalPosition = initialPos;
 			}
 
-			Rotation = Vector3.Zero;
```
Like Robert said, it's not redundant. The Box's Rotation needs to be set back to its original value. However, there is a bug. Zero might not be the original Rotation.

Changing `GlobalPosition = initialPos` to `GlobalTranform = initialTransform` would fix that.

cooperra replied:
The same fix also need to be applied to Pallet.cs.

ryevdokimov commented on src/Box/Box.cs:
```diff
@@ -100,20 +102,18 @@ void OnSimulationEnded()
 			{
-				GlobalPosition = initialPos;
+				GlobalTransform = initialTransform;
+				Rotation = initialRotation; // Reset to initial rotation
```
This is redundant, an object's transform contains its rotation. 

See: https://docs.godotengine.org/en/4.3/classes/class_transform3d.html

ryevdokimov commented on src/Pallet/Pallet.cs:
```diff
@@ -101,20 +103,18 @@ void OnSimulationEnded()
 			{
-				GlobalPosition = initialPos;
+				GlobalTransform = initialTransform;
+				Rotation = initialRotation; // Reset to initial rotation
```
Same here.

ryevdokimov commented on src/Pallet/Pallet.cs:
```diff
@@ -101,20 +103,18 @@ void OnSimulationEnded()
 			rigidBody.AngularVelocity = Vector3.Zero;
 
-			//Work around for #83 
+			//Work around for #S83 
```
Not sure what the added `S` implies here. If it is accidental, please remove it.

Mareh07 replied:
typo, I missed it. I will remove it.

Mareh07 commented:
I have cleaned up the code but I am not entirely sure if I fully understood the feedback. In the end, I decided to remove:

`Rotation = Vector3.Zero;`

As @cooperra mentioned, this resets the box to zero rotation and not initial one. To address this I added:

`initialRotation = Rotation;`

and encapsulated the reset within the if condition because I thought that was its intended purpose:

```
// Workaround for #S83
if (enable_inital_pos)
{
    GlobalTransform = initialTransform;
    Rotation = initialRotation; // Reset to initial rotation
}
```

Please review and let me know if I missed anything or misunderstood the feedback. Thank you for your support!

ryevdokimov commented on src/Box/Box.cs:
```diff
@@ -100,20 +102,18 @@ void OnSimulationEnded()
 			rigidBody.AngularVelocity = Vector3.Zero;
 
 			//Work around for #83 
 			if (enable_inital_pos)
```
Rename the variable in this condition to `enable_inital_trasform`, so that it matches the action it encloses.

ryevdokimov commented:
Looks good, can you squash the commits and rename to something like the title?

Mareh07 commented:
I've squashed the commits and updated the PR as requested. Please let me know if there's anything else I should address.

ryevdokimov commented:
Thanks!
